### PR TITLE
Slightly simplify myjsonrpc

### PR DESCRIPTION
**Pull request by okurz**
https://progress.opensuse.org/issues/167935

Potential further step: https://github.com/okurz/os-autoinst/tree/feature/simplify_extract_loop

Related to
* #2578

### CODE REVIEW

File: myjsonrpc.pm [hunk 1]
```diff
@@ -74,10 +70,7 @@ sub read_json ($socket, $cmd_token = undef, $multi = undef) {
         # remember the trailing text
         if ($hash) {
             $sockets->{$fd} = $cjx->incr_text();
-            if (DEBUG_JSON || $bmwqemu::vars{DEBUG_JSON_RPC}) {
-                my $token = $hash->{json_cmd_token} // 'no-token';
-                bmwqemu::diag("read_json($fd) json_cmd_token=$token");
-            }
+            bmwqemu::diag("read_json($fd) json_cmd_token=" . $hash->{json_cmd_token} // 'no-token') if is_debug;
```
Comment (Martchus) [inline myjsonrpc.pm:73]:
```suggestion
            bmwqemu::diag("read_json($fd) json_cmd_token=" . ($hash->{json_cmd_token} // 'no-token')) if is_debug;
```

File: myjsonrpc.pm [hunk 2]
```diff
@@ -95,22 +88,16 @@ sub read_json ($socket, $cmd_token = undef, $multi = undef) {
...
+            bmwqemu::diag("read_json($fd): can_read's underlying system call has been interrupted, trying again\n") if is_debug;
         }
 
         my $qbuffer;
-        my $bytes = sysread($socket, $qbuffer, READ_BUFFER);
-        if (!$bytes) {
-            bmwqemu::fctwarn("sysread failed: $!") if DEBUG_JSON || $bmwqemu::vars{DEBUG_JSON_RPC};
-            return;
-        }
+        my $bytes = sysread($socket, $qbuffer, READ_BUFFER) or do { bmwqemu::fctwarn("sysread failed: $!") if is_debug and return undef };
```
Comment (Martchus) [inline myjsonrpc.pm:100]: This is really hard to read. It is really unclear what the left-hand-side of the `and` evaluates to if `is_debug` is false and it also looks like `is_debug and return undef` is all just part of the `if` condition although the `and return undef` part is actually not (meant) to be part of it.

I think the following will work (provided `bmwqemu::fctwarn` returns undef as well):
```suggestion
        my $bytes = sysread($socket, $qbuffer, READ_BUFFER) or return (is_debug ? bmwqemu::fctwarn("sysread failed: $!") : undef;
```

**Review by Martchus — CHANGES_REQUESTED** [review]

File: myjsonrpc.pm (hunk 2 above)
Comment (perlpunk) [reply myjsonrpc.pm:100]: But if we have a `do` block there's no reason to just use multiple statements separated by `;`.

Comment (perlpunk) [reply myjsonrpc.pm:100]: I don't have a problem with the original code, and putting it in one line is also just cheating with code coverage, and explicitly marking those lines as uncoverable would make it more clear that this **is** actually uncovered.

**Comment by Wabri** [comment]
I've cherry-picked during my work on #2578 and than mixed with the suggestions given by @Martchus => https://github.com/os-autoinst/os-autoinst/pull/2578/commits/fb19b3c34dfec2dc4d46b622e701f77c333da5c7

**Pull request closed without merging**